pkg/northbound/gnmi/subscribe.go

```diff
@@ -148,7 +149,12 @@ func broadcastNotification() {
 				//FIXME this might prove expensive, find better way to store subscriptionPath and target in channels map
 				subscriptionPathStr := utils.StrPath(subscriptionPath)
 				if subscriptionPath.Target == target && strings.HasPrefix(changeValue.Path, subscriptionPathStr) {
-					sendUpdate(subscriptionChan, subscriptionPath, changeValue.Value)
+					pathGnmi, err := utils.ParseGNMIElements(utils.SplitPath(changeValue.Path))
+					if err != nil {
+						log.Println("Error in parsing path", err)
+					}
+					pathGnmi.Target = subscriptionPath.Target
```

Comment:
if you got an error 2 lines up, then you might get a segv assigning to Target of pathGnmi here